homeassistant/components/pyload/sensor.py

```diff
@@ -109,22 +128,36 @@ def __init__(
         self.type = sensor_type.key
         self.api = api
         self.entity_description = sensor_type
+        self.data: StatusServerResponse
 
-    def update(self) -> None:
+    async def async_update(self) -> None:
         """Update state of sensor."""
         try:
-            self.api.update()
-        except requests.exceptions.ConnectionError:
-            # Error calling the API, already logged in api.update()
-            return
-
-        if self.api.status is None:
-            _LOGGER.debug(
-                "Update of %s requested, but no status is available", self.name
-            )
-            return
+            self.data = await self.api.get_status()
+        except InvalidAuth:
+            _LOGGER.info("Authentication failed, trying to reauthenticate")
+            try:
+                await self.api.login()
```

Comment:
It seems like this retries the login, but does not repeat the `get_status` call which originally failed. Maybe it would be cleaner if the external package performed the login internally such that clients need not worry about this retry logic at all.

Reply:
the next polling happens like 15 seconds later anyway, but you are right, i should either raise UpdateFailed, so that HA would show on the next polling that it recovered or call get_status again directly after the login

Reply:
i decided for raising UpdateFailed and letting the next polling retry to fetch the data, calling get_status again after login would also mean having to catch again all the exceptions